bundles/org.openhab.core.automation.rest/src/main/java/org/openhab/core/automation/rest/internal/ThingActionsResource.java

```diff
@@ -238,13 +264,25 @@ public Response executeThingAction(@PathParam("thingUID") @Parameter(description
         return scopeAnnotation.name();
     }
 
+    private boolean hasInput(ActionType actionType, String in) {
```

Comment:
Can be replaced by a one-liner: `actionType.getInputs().stream().anyMatch(i -> i.getName().equals(in);`. This is probably alsio faster because it can be executed in parallel.

Reply:
Replaced